### PR TITLE
chore: added trufflehog github action for secrets scanning

### PR DESCRIPTION
**Pull request by olcuhu**
**All Submissions.**

- [X] Have you followed the guidelines in our [Contributing document](../../CONTRIBUTING.md)?
- [X] Have you checked to ensure there aren't other open [Pull Requests][pull_requests] for the same update/change?

**Code Submissions.**

- [X] Does your submission pass linting, tests, and security analysis?

**Changes to Core Features.**

- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
Including Trufflehog github action into the security_analysis workflow for secrets detection in PRs, as mentioned in #129.
- [X] Have you written new tests for your core changes, as applicable?
n/a

### CODE REVIEW

File: .github/workflows/security_analysis.yml [hunk 1]
```diff
@@ -48,3 +48,19 @@ jobs:
         uses: github/codeql-action/upload-sarif@v1
         with:
           sarif_file: trivy-results.sarif
+
+  TruffleHog:
```
Comment (06kellyjac) [inline .github/workflows/security_analysis.yml:52]:
```suggestion
  trufflehog:
```

File: .github/workflows/security_analysis.yml [hunk 2]
```diff
@@ -48,3 +48,19 @@ jobs:
...
+    name: TruffleHog
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: TruffleHog OSS
```
Comment (06kellyjac) [inline .github/workflows/security_analysis.yml:60]:
```suggestion
      - name: Run Trufflehog
```

**Comment by 06kellyjac** [comment]
can you squash this to 1 commit please. then we can merge

**Comment by olcuhu** [comment]
> can you squash this to 1 commit please. then we can merge

done! managed to keep your co-authored commit this time too :joy:

**Review by 06kellyjac — APPROVED** [review]

**Pull request merged**